index.js

```diff
@@ -191,6 +192,15 @@ Prompt.prototype.search = function(searchTerm) {
       return choice.type !== 'separator';
     }));
 
+    if (self.firstRender) {
+      // If default is a Number, then use as index. Otherwise, check for value.
+      if (typeof self.default === 'number' && self.default >= 0 && self.default < choices.realLength) {
+        self.selected = self.default;
+      } else if (typeof self.default !== 'number' && self.default != null) {
```

Comment:
I think this should not be !== 'number', but rather === 'string', since we should support function as well.

I guess a default function should be called at init, but maybe also after each search? With arguments: answersSoFar and returnedResults